walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,46 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    ArrayList<TimeRange> possibleMeetingTimes = new ArrayList<>();
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return possibleMeetingTimes;
+    } 
+
+    possibleMeetingTimes.add(TimeRange.WHOLE_DAY);
+
+    for (Event event : events) {
+      if (!Collections.disjoint(event.getAttendees(), attendees)) {
```

Comment:
nit: you can reduce some of this nesting by flipping the if condition and using `continue`